e2e/mock/server.go

```diff
@@ -3,39 +3,56 @@ package mock
 import (
 	"net"
 
-	"github.com/hyperledger/fabric-protos-go/orderer"
-	"github.com/hyperledger/fabric-protos-go/peer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
 type Server struct {
-	GrpcServer *grpc.Server
-	Listener   net.Listener
+	grpcServers []*grpc.Server
+	Listeners   []net.Listener
+	peers       []*Peer
+	orderer     *Orderer
+	Credentials credentials.TransportCredentials
 }
 
-func (s *Server) Start() {
+func (s *Server) Prepare() {
```

Comment:
this method, as well as `peer.Prepare` & `orderer.Prepare`, _looks like_ constructors, why don't we just make them constructors? something like:
```go
func NewServer(peerN int) *Server
```
and add a method
```go
func (s *Server) Addresses() (peerAddrs []string, ordererAddr string)
```

Reply:
and if i understand correctly, user (of mock.Server) doesn't really care about `listeners`, right?